### DISCUSSION TITLE
Runtime of the pipeline

### DISCUSSION THREAD

**Discussion by sgibb** [ropensci/targets#801] [Help] [answered]
In `tar_visnetwork` or `tar_meta` I could see the runtime of single targets/branches but is there a way to get the total runtime? Not just of the current/last run but of the whole pipeline if it where run the first time?

Currently I am using the following but it doesn't seem to report the correct time (e.g. it reports ~ 2 hours but I have a branch that runs for ~ 30 min on 32 cores which should return something like 48 hours, also it seems `tar_meta` stores some information about (dynamic) branches that are run days before (with different settings/input):
```r
elapsed_runtime <- sum(
    subset(tar_meta(), !type %in% c("function", "object", "branch"))$seconds
)
```

**Comment by wlandau** [comment, marked as answer]
Total runtime assuming no targets are skipped? This depends on [scheduling and parallel computing](http://books.ropensci.org/targets/hpc.html), so different initial runs on different machines will give different times. Summing the times like you are already doing gives you a decent estimate of the purely sequential case. 

If you have a single branch runs for 30 minutes, `targets` will store an elapsed runtime of 30 minutes regardless of how many cores you use within that target (e.g. via tar_target(x, parallel::mclapply(..., mc.cores = 32))`.

**Reply by sgibb** [reply]
Ok, thanks!
